translator/src/main/java/com/google/devtools/j2objc/javac/JavacParser.java

```diff
@@ -235,13 +243,14 @@ public ProcessingResult processAnnotations(Iterable<String> fileArgs,
     PathClassLoader loader = new PathClassLoader(options.fileUtil().getClassPathEntries());
     loader.addPaths(options.getProcessorPathEntries());
     Iterator<Processor> serviceIterator = ServiceLoader.load(Processor.class, loader).iterator();
-    if (serviceIterator.hasNext()) {
+    String explicitProcessorList = options.getProcessors();
+    if (serviceIterator.hasNext() || explicitProcessorList != null) {
```

Comment:
Revert. Most annotation processing jars define a ServiceLoader entry in their manifest files, which is why this issue wasn't caught sooner. This test therefore needs to remain, and support for your use case is handled in getJavacOptions().

Reply:
@tomball - We have two types of processors here:
1) ServiceLoader processors (captured by serviceIterator.hasNext())
2) -processor processors

If we don't check for options.getProcessors() here and there are no service processors present, we'd never run the preprocessing step, even if we specify the -processor flag. This is why I'm currently checking both here and running pre-processing if either is present (which tests corroborate).

There is a slight change to the way j2objc currently operate, as the -processor flag is currently ignored and will now override any ServiceLoader processor, due to the interface defined by javac (and also documented in the j2objc -help of the -processor flag)